src/Analyzer/EtSchemaAnalyzer.php

```diff
@@ -0,0 +1,367 @@
+            array_diff(
+                array_keys($beforeRecord['field']),
+                array_keys($afterRecord['field'])
+            ),
+            array_diff(
+                array_keys($afterRecord['field']),
+                array_keys($beforeRecord['field'])
+            )
+        );
+        foreach ($diff as $fieldName) {
+            if (isset($beforeRecord['field'][$fieldName])) {
+                $changes[] = $this->removedField($moduleName, $beforeRecord['name'], $fieldName);
+            } else {
+                $changes[] = $this->addedField($moduleName, $afterRecord['name'], $fieldName);
+            }
+        }
+        return $changes;
+    }
+
+    /**
+     * Analyze module configuration file
+     *
+     * @param string $moduleName
+     * @param array $beforeModuleConfig
+     * @param array $afterModuleConfig
+     * @return array
+     */
+    private function analyzeModuleConfig(string $moduleName, array $beforeModuleConfig, array $afterModuleConfig): array
+    {
+        $changes = [];
+        $commonRecords = array_intersect(
+            array_keys($beforeModuleConfig),
+            array_keys($afterModuleConfig)
+        );
+        foreach ($commonRecords as $recordName) {
+            $changes = array_merge(
+                $changes,
+                $this->analyzeRecord(
+                    $moduleName,
+                    $beforeModuleConfig[$recordName],
+                    $afterModuleConfig[$recordName]
+                )
+            );
+        }
+        $diff = array_merge(
+            array_diff(
+                array_keys($beforeModuleConfig),
+                array_keys($afterModuleConfig)
+            ),
+            array_diff(
+                array_keys($afterModuleConfig),
+                array_keys($beforeModuleConfig)
+            )
+        );
+        foreach ($diff as $recordName) {
+            if (isset($beforeModuleConfig[$recordName])) {
+                $changes[] = $this->removedRecord($moduleName, $recordName);
+            } else {
+                $changes[] = $this->addedRecord($moduleName, $recordName);
+            }
+        }
+        return $changes;
+    }
+
+    /**
+     * Register changes to the report
+     *
+     * @param array $changes
+     */
+    public function reportChanges(array $changes): void
+    {
+        foreach ($changes as $change) {
+            $this->report->add(
+                self::CONTEXT,
+                new EtSchemaOperation(
+                    $change['location'],
+                    $change['code'],
+                    $change['target'],
+                    $change['reason'],
+                    $change['level']
+                )
+            );
+        }
+    }
+
+    /**
+     * Analyze configuration changes
+     *
+     * @param Stmt|Registry $registryBefore
+     * @param Stmt|Registry $registryAfter
+     * @return Report
+     */
+    public function analyze($registryBefore, $registryAfter)
```

Comment:
return type missing

Reply:
as design, a declaration inherited from the abstract declaration